content/automate/ManageIQ/System/Event/StateMachines/Refresh.class/__methods__/target_refresh.rb

```diff
@@ -0,0 +1,50 @@
+#
+# Description: This method starts a synchronous refresh and save the
+#              task id in the state variables so we can use it when
+#              waiting for the refresh to finish.
+#
+
+module ManageIQ
+  module Automate
+    module System
+      module Event
+        module StateMachines
+          module Refresh
+            class TargetRefresh
+              def initialize(handle = $evm)
+                @handle = handle
+              end
+
+              def main
+                task_id = event.refresh(refresh_target, true)
+                @handle.set_state_var(:refresh_task_id, task_id)
```

Comment:
@lfu you mentioned that the task_id can be nil if there are no targets, should we be stopping the state machine with an error if there are no task ids instead of storing a nil